frameworks/Swift/swift-nio/Sources/swift-nio-tfb-default/main.swift

```diff
@@ -0,0 +1,114 @@
+	static let plainTextResponseLengthString = String(plainTextResponseLength)
+
+	static let jsonResponseLength = try! JSONEncoder().encode(JSONTestResponse()).count
+	static let jsonResponseLengthString = String(jsonResponseLength)
+}
+
+private final class HTTPHandler: ChannelInboundHandler {
+
+	public typealias InboundIn = HTTPServerRequestPart
+	public typealias OutboundOut = HTTPServerResponsePart
+
+	let dateFormatter = RFC1123DateFormatter()
+	let jsonEncoder = JSONEncoder()
+
+	func channelRead(ctx: ChannelHandlerContext, data: NIOAny) {
+		let reqPart = self.unwrapInboundIn(data)
+
+		switch reqPart {
+		case .head(let request):
+			switch request.uri {
+			case "/plaintext":
+				processPlaintext(ctx: ctx)
+			case "/json":
+				processJSON(ctx: ctx)
+			default:
+				_ = ctx.close()
+			}
+		case .body:
+			break
+		case .end:
+			_ = ctx.writeAndFlush(self.wrapOutboundOut(.end(nil)), promise: nil)
+		}
+	}
+
+	private func processPlaintext(ctx: ChannelHandlerContext) {
+		let responseHead = plainTextResponseHead(contentLength: Constants.plainTextResponseLengthString)
+		ctx.write(self.wrapOutboundOut(.head(responseHead)), promise: nil)
+
+		var bodyBuffer = ctx.channel.allocator.buffer(capacity: Constants.plainTextResponseLength)
+		bodyBuffer.write(staticString: Constants.plainTextResponse)
+		ctx.write(self.wrapOutboundOut(.body(.byteBuffer(bodyBuffer))), promise: nil)
+	}
+
+	private func processJSON(ctx: ChannelHandlerContext) {
+		let responseHead = jsonResponseHead(contentLength: Constants.jsonResponseLengthString)
+		ctx.write(self.wrapOutboundOut(.head(responseHead)), promise: nil)
+
+		var bodyBuffer = ctx.channel.allocator.buffer(capacity: Constants.jsonResponseLength)
+		let responseData = try! jsonEncoder.encode(JSONTestResponse())
+		bodyBuffer.write(bytes: responseData)
+		ctx.write(self.wrapOutboundOut(.body(.byteBuffer(bodyBuffer))), promise: nil)
+	}
+
+	private func jsonResponseHead(contentLength: String) -> HTTPResponseHead {
+		return responseHead(contentType: "application/json", contentLength: contentLength)
+	}
+
+	private func plainTextResponseHead(contentLength: String) -> HTTPResponseHead {
+		return responseHead(contentType: "text/plain", contentLength: contentLength)
+	}
+
+	private func responseHead(contentType: String, contentLength: String) -> HTTPResponseHead {
+		var headers = HTTPHeaders()
+		headers.replaceOrAdd(name: "content-type", value: contentType)
+		headers.replaceOrAdd(name: "content-length", value: contentLength)
+		headers.replaceOrAdd(name: "server", value: Constants.serverName)
+		headers.replaceOrAdd(name: "date", value: dateFormatter.getDate())
+
+		return HTTPResponseHead(version: Constants.httpVersion,
+								status: .ok,
+								headers: headers)
+	}
+}
+
+let group = MultiThreadedEventLoopGroup(numThreads: System.coreCount)
+let bootstrap = ServerBootstrap(group: group)
+	.serverChannelOption(ChannelOptions.socket(SocketOptionLevel(SOL_SOCKET), SO_REUSEPORT), value: 1)
```

Comment:
I don't think you're getting anything from setting `SO_REUSEPORT` here: you only ever bind this socket once.